src/engine/TextIndexScanForWord.cpp

```diff
@@ -114,6 +114,9 @@ string TextIndexScanForWord::getCacheKeyImpl() const {
   std::ostringstream os;
   os << "WORD INDEX SCAN: "
      << " with word: \"" << config_.word_ << "\"";
+  if (config_.scoreVar_.has_value()) {
+    os << " and scoreVar: \"" << config_.scoreVar_.value().name() << "\"";
+  }
```

Comment:
Please add a unit test for this in `test/engine/TextIndexScanForWordTest.cpp`, there are already some other tests for cache keys.